tests/10-api.spec.js

```diff
@@ -1,18 +1,292 @@
+    it('should succeed if correct data is passed', async () => {
+      const url = 'http://localhost:5000/documents';
+      // Admin seed
+      const seed = 'z1AZK4h5w5YZkKYEgqtcFfvSbWQ3tZ3ZFgmLsXMZsTVoeK7';
+      const decoded = decodeSecretKeySeed({secretKeySeed: seed});
+
+      const {methodFor} = await didKeyDriver.generate({seed: decoded});
+      const invocationCapabilityKeyPair = methodFor(
+        {purpose: 'capabilityInvocation'});
+
+      const zcapClient = new ZcapClient({
+        SuiteClass: Ed25519Signature2020,
+        invocationSigner: invocationCapabilityKeyPair.signer()
+      });
+      let res;
+      let err;
+      try {
+        res = await zcapClient.write({url, json: {name: 'test'}});
+      } catch(e) {
+        err = e;
+      }
+      should.exist(res);
+      should.not.exist(err);
+      res.status.should.equal(200);
+      res.data.message.should.equal('Post request was successful.');
+    });
+    it('should error if missing authorization header', async () => {
+      let res;
+      let err;
+      try {
+        res = await httpClient.post('http://localhost:5000/documents', {
+          json: {}
+        });
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(res);
+      should.exist(err);
+      err.status.should.equal(500);
+      err.data.name.should.equal('DataError');
+      err.data.message.should.equal(
+        'Missing or invalid "authorization" header.');
+    });
+    it('should throw forbidden error if the authorized invoker does not ' +
+      'match the controller', async () => {
+      const url = 'http://localhost:5000/documents';
+      // Use a different seed
+      const seed = 'z1AbCFiBWpN89ug5hcxUfa6TzpGoowH7DBidgL8zPu6v5RV';
+      const decoded = decodeSecretKeySeed({secretKeySeed: seed});
+
+      const {methodFor} = await didKeyDriver.generate({seed: decoded});
+      const invocationCapabilityKeyPair = methodFor(
+        {purpose: 'capabilityInvocation'});
+
+      const zcapClient = new ZcapClient({
+        SuiteClass: Ed25519Signature2020,
+        invocationSigner: invocationCapabilityKeyPair.signer()
+      });
+      let res;
+      let err;
+      try {
+        res = await zcapClient.write({url, json: {name: 'test'}});
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(res);
+      should.exist(err);
+      err.status.should.equal(403);
+      err.message.should.equal('Forbidden');
+    });
+    it('should throw error if digest header is not present when HTTP body is ' +
+      'present', async () => {
+      const url = 'http://localhost:5000/documents';
+      // Use a different seed
+      const seed = 'z1AbCFiBWpN89ug5hcxUfa6TzpGoowH7DBidgL8zPu6v5RV';
+      const decoded = decodeSecretKeySeed({secretKeySeed: seed});
+
+      const {methodFor} = await didKeyDriver.generate({seed: decoded});
+      const invocationCapabilityKeyPair = methodFor(
+        {purpose: 'capabilityInvocation'});
```

Comment:
This is repeated a lot -- it should be a helper function.

Reply:
> Any reason we didn't use `CapabilityAgent` here from webkms-client? 

Didn't use `CapabilityAgent` because I didn't know we could use that here as I was given the above example to follow. 

> However, if we create a helper function for this that just takes a `seed` and returns an `invocationSigner` (e.g., `helpers.getInvocationSigner({seed: <encodedSeed>})`) we can easily swap either later.

Added helper function here d867de7